### PR TITLE
Fix player controlled beepskys cuffing people

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

 
When player controlled beepskys tried to cuff people it would just bug out. THIS fixes it

## Why It's Good For The Game

Es

## Changelog

:cl:
fix: Cuffing people as a PAI/ghost controlled beepsky no longer glitches out
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/bot/secbot.dm [hunk 1]
```diff
@@ -276,21 +276,18 @@
...
-	if(!isturf(current_target.loc))
-		return
-	if(!Adjacent(current_target))
+	if((!(bot_mode_flags & BOT_MODE_ON)) || !isturf(current_target.loc) || !Adjacent(current_target) || current_target.handcuffed  ) //if he's in a closet or not adjacent, we cancel cuffing.
 		return
+	current_target.set_handcuffed(new /obj/item/restraints/handcuffs/cable/zipties/used(current_target))
+	current_target.update_handcuffed()
+	playsound(src, "law", 50, FALSE)
+	back_to_idle()
+	return TRUE
```
Comment (JohnFulpWillard) [inline code/modules/mob/living/simple_animal/bot/secbot.dm:288]: what is this returning for?

File: code/modules/mob/living/simple_animal/bot/secbot.dm [hunk 2]
```diff
@@ -276,21 +276,18 @@
...
-	addtimer(CALLBACK(src, .proc/handcuff_target, target), 6 SECONDS)
+	addtimer(CALLBACK(src, .proc/handcuff_target, current_target), 6 SECONDS)
 
 /mob/living/simple_animal/bot/secbot/proc/handcuff_target(mob/living/carbon/current_target)
-	if(!(bot_mode_flags & BOT_MODE_ON)) //if he's in a closet or not adjacent, we cancel cuffing.
-		return
-	if(!isturf(current_target.loc))
-		return
-	if(!Adjacent(current_target))
+	if((!(bot_mode_flags & BOT_MODE_ON)) || !isturf(current_target.loc) || !Adjacent(current_target) || current_target.handcuffed  ) //if he's in a closet or not adjacent, we cancel cuffing.
```
Comment (JohnFulpWillard) [inline code/modules/mob/living/simple_animal/bot/secbot.dm:282]: i split this into 3 separate checks for readability

Comment (SmoSmoSmoSmok) [reply code/modules/mob/living/simple_animal/bot/secbot.dm]: > i split this into 3 separate checks for readability

ok

**Review by JohnFulpWillard — APPROVED** [review]

**Comment by Shadowflame909** [comment]
I thought you couldn't put pais into sec-trons

**Comment by ATH1909** [comment]
> I thought you couldn't put pais into sec-trons

you can't, but you can still get player-controlled secbots from things like the random sentience event

**Comment by ATH1909** [comment]
please change the title of this pr, you spooked me into believing that this pr was REMOVING the ability of player-controlled secbots to cuff people

**Comment by Mothblocks** [comment]
What ath said

**Comment by ATH1909** [comment]
> What ath said

the title still makes it sound like he's "fixing" player-controlled secbots being able to cuff people by removing that feature

**Pull request merged**